src/material/schematics/ng-update/migrations/legacy-components-v15/index.ts

```diff
@@ -6,12 +6,69 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {Migration, TargetVersion} from '@angular/cdk/schematics';
 import * as ts from 'typescript';
+import * as postcss from 'postcss';
+import * as scss from 'postcss-scss';
+
+import {MIXINS} from './constants';
+
+import {Migration, ResolvedResource, TargetVersion, WorkspacePath} from '@angular/cdk/schematics';
 
 export class LegacyComponentsMigration extends Migration<null> {
   enabled = this.targetVersion === TargetVersion.V15;
 
+  override visitStylesheet(stylesheet: ResolvedResource): void {
+    let namespace: string | undefined = undefined;
+    const processor = new postcss.Processor([
+      {
+        postcssPlugin: 'legacy-components-v15-plugin',
+        AtRule: {
+          use: node => {
+            namespace = namespace ?? this._parseSassNamespace(node);
+          },
+          include: node => this._handleAtInclude(node, stylesheet.filePath, namespace),
+        },
+      },
+    ]);
+    processor.process(stylesheet.content, {syntax: scss}).sync();
+  }
+
+  /** Returns the namespace of the given at-rule if it is importing from @angular/material. */
+  private _parseSassNamespace(node: postcss.AtRule): string | undefined {
+    if (node.params.startsWith('@angular/material', 1)) {
+      return node.params.split(/\s/).pop();
+    }
+    return;
+  }
+
+  /** Handles updating the at-include rules of legacy component mixins. */
+  private _handleAtInclude(
+    node: postcss.AtRule,
+    filePath: WorkspacePath,
+    namespace?: string,
+  ): void {
+    if (!namespace || !node.source?.start) {
+      return;
+    }
+    const startStr = namespace + '.';
+    if (this._isLegacyMixin(node, namespace)) {
+      this._replaceAt(filePath, node.source.start.offset, {
+        old: startStr,
+        new: namespace + '.legacy-',
+      });
+    }
+  }
+
+  /** Returns true if the given at-include rule is a use of a legacy component mixin. */
+  private _isLegacyMixin(node: postcss.AtRule, namespace: string): boolean {
+    for (let i = 0; i < MIXINS.length; i++) {
+      if (node.params.startsWith(namespace + '.' + MIXINS[i])) {
```

Comment:
nit/opt: 
```
`${namespace}.${MIXINS[i]}`
```

Reply:
actually you're not using the index, so can do `for (const mixin of MIXINS` as well